int64 state
int64 inter_node
int64 cur_node